frontend/src/concepts/roleBinding/RoleBindingPermissionsNameInput.tsx

```diff
@@ -68,6 +64,18 @@ const RoleBindingPermissionsNameInput: React.FC<RoleBindingPermissionsNameInputP
         }
       }}
       placeholder={placeholderText}
+      createOptionMessage={(newValue) => `Select "${newValue}"`}
+      filterFunction={(filterValue, options) => {
```

Comment:
Added this filterFunction in order to achieve the [requirement](https://issues.redhat.com/browse/RHOAIENG-12025#:~:text=It%20should%20act%20like%20this%20example%20(but%20with%20Select%20instead%20of%20Use)%20%2D%20if%20the%20user%20selects%20a%20custom%20item%20like%20this%2C%20then%20changes%20their%20selection%2C%20the%20custom%20item%20should%20disappear.) It filters the existing options based on the input and adds a new option with the exact input value if no matches are found.

Reply:
I don't see any different behavior here when I try removing this `filterFunction`. With or without it, I'm actually seeing that if I click the new option that doesn't match any select items, it doesn't remain selected in the dropdown. I think that's because we still need it to be one of the `selectOptions` passed above for it to show as selected.

I also notice that with this `filterFunction` in place, if you type text that doesn't even partially match any other options, it gets shown as the sole option without the "Select ___" text. So I think we don't want this `filterFunction` and we need to fix the issue some other way.

Reply:
@YuliaKrimerman I think I figured it out - Try this. Remove this `filterFunction`, and above where you removed the other code, add this:

```ts
  // If we've selected an option that doesn't exist via isCreatable, include it in the options so it remains selected
  if (!selectOptions.some((option) => option.value === value)) {
    selectOptions.push({ value, content: value });
  }
```

That ensures that our "newly created" item is in the list once it is selected, but disappears from the list if it is deselected.